shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -106,6 +106,7 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool)
     TypeDef(Cls, TypeName("true"), Nil, Nil, TopType, Nil, Nil, Set.single(Var("bool")), N) ::
     TypeDef(Cls, TypeName("false"), Nil, Nil, TopType, Nil, Nil, Set.single(Var("bool")), N) ::
     TypeDef(Cls, TypeName("string"), Nil, Nil, TopType, Nil, Nil, Set.empty, N) ::
+    TypeDef(Cls, TypeName("undefined"), Nil, Nil, TopType, Nil, Nil, Set.empty, N) ::
```

Comment:
Actually this makes `undefined` a normal nominal class, but it's not – it's a literal class. It should be:
```suggestion
    TypeDef(Als, TypeName("undefined"), Nil, Nil, ClassTag(UnitLit(true), Set.empty)(noProv), Nil, Nil, Set.empty, N) ::
```